pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go

```diff
@@ -199,12 +199,16 @@ KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/k
 .PHONY: kustomize
 kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
 $(KUSTOMIZE): $(LOCALBIN)
+	@if test -x $(LOCALBIN)/kustomize && ! $(LOCALBIN)/kustomize version | grep -q $(KUSTOMIZE_VERSION); then
+		echo "$(LOCALBIN)/kustomize version is not expected $(KUSTOMIZE_VERSION). Removing it before installing.";
+		rm -rf $(LOCALBIN)/kustomize;
+	fi
 	test -s $(LOCALBIN)/kustomize || { curl -Ss $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
 
 .PHONY: controller-gen
 controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
 $(CONTROLLER_GEN): $(LOCALBIN)
-	test -s $(LOCALBIN)/controller-gen || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
+	test -s $(LOCALBIN)/controller-gen && $(LOCALBIN)/controller-gen --version | grep -q $(CONTROLLER_TOOLS_VERSION) || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
 
 .PHONY: envtest
```

Comment:
Also, you need run make generate to update all samples within

Reply:
I am not sure what you mean by this. Running `make generate` has the `controller-gen` target as a prerequisite.

Reply:
@coriaedu I believe what @camilamacedo86 is referring to is for generating our testdata samples. The Makefile at the root of the Kubebuilder project should have a target called `generate` that will automatically regenerate the sample operators under the `testdata/` directory with any changes that have been made to the scaffolding. 

Running this `make generate` from the root of the project directory is important so that we can test that the changes to the scaffolding logic/templates are accurately done. We also use the operators in this `testdata/` directory to run e2e tests to ensure that scaffolding changes don't break the expected workflow.

Reply:
I will give that all a try! Thanks @everettraven for the detailed explanation and @camilamacedo86 for the prompt review. I am excited to be contributing to this project!

Reply:
Done